myapp/app/controllers/application_controller.rb

```diff
@@ -1,2 +1,38 @@
 class ApplicationController < ActionController::Base
+
+  around_action :switch_locale
+
+  def switch_locale(&action)
+    locale = params[:locale] || I18n.default_locale
+    I18n.with_locale(locale, &action)
+  end
+
+  # keep locale in urls
+  def default_url_options
+    { locale: I18n.locale }
+  end
+
+
+  # unless Rails.env.development?
+    rescue_from Exception,                        with: :_render_500
```

Comment:
It is generally not recommended to rescue from Exception, as it may hide critical system errors. Instead, the default behavior of Rails is preferred, which will handle exceptions and display an error page in production mode.

For 500 errors you can maybe rescue ActiveRecord::StatementInvalid. If a database query fails due to syntax errors or other issues, the application will catch the exception, log the error, and render an appropriate 500 error page for the user.

Reply:
I see. Thank you so much.
So what about other exceptions rather than `ActiveRecord::StatementInvalid`? 
Errors other than Database related which are belongs to 500, should i rescue_from another entity?

Reply:
Of course, you can rescue those as well. I just gave an example.